### PR TITLE
ADD contract_journal module

### PR DESCRIPTION
**Pull request by angelmoya**

### CODE REVIEW

File: contract_journal/README.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+========================
+Contract Account Journal
```
Comment (pedrobaeza) [inline contract_journal/README.rst]: Contract Invoice Journal

File: contract_journal/README.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+========================
+
+This module was written to extend the functionality of contracts to support to
+set an account journal on contract.
```
Comment (pedrobaeza) [inline contract_journal/README.rst]: This module allows to set a sales account journal for creating the invoices on it.

File: contract_journal/models/contract.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            ('company_id', '=', company_id)]
+        return self.env['account.journal'].search(domain, limit=1)
+
+    journal_id = fields.Many2one(
+        'account.journal',
+        string='Journal',
+        default=_default_journal,
+        domain="[('type', '=', 'sale'),('company_id', '=', company_id)]")
+
+    def _prepare_invoice_data(self, cr, uid, contract, context=None):
```
Comment (pedrobaeza) [inline contract_journal/models/contract.py]: Why don't put `@api.model` and you avoid old API?

File: contract_journal/models/contract.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        context = context or {}
+        invoice = super(AccountAnalyticAccount, self)._prepare_invoice_data(
```
Comment (pedrobaeza) [inline contract_journal/models/contract.py]: s/invoice/invoice_vals (to highlight that it's a dictionary with vals, not a recordset)

File: contract_journal/models/contract.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+            cr, uid, contract, context=context)
+        invoice.update({
```
Comment (pedrobaeza) [inline contract_journal/models/contract.py]:
```
if contract.journal_id:
    invoice_vals['journal_id'] = contract.journal_id.id
```

File: contract_journal/models/contract.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    @api.model
+    def _default_journal(self):
+        company_id = self._context.get(
```
Comment (pedrobaeza) [inline contract_journal/models/contract.py]: Use `self.env.context`, not private variable.

**Comment by angelmoya** [comment]
Ok @pedrobaeza , work done

**Comment by pedrobaeza** [comment]
Thanks! Maybe an improvement in tests is to create another journal, because sales_journal is the default one, so there's no possibility to check if the journal is set because of default or the module code.

Anyway, :+1:

**Comment by angelmoya** [comment]
Thanks @pedrobaeza , test are doing with sales_journal and with other custom journal that is created in the test.

**Comment by pedrobaeza** [comment]
Oh, sorry, I didn't see it. Perfect then!

**Comment by dreispt** [comment]
:+1: +1:

**Pull request merged**